iggtools/common/utils.py

```diff
@@ -438,7 +438,11 @@ def flatten(l):
 
 @retry
 def upload(src, dst, check=True):
-    command(f"set -o pipefail; lz4 -c {src} | aws s3 cp --only-show-errors - {dst}", check=check)
+    try:
+        command(f"set -o pipefail; lz4 -c {src} | aws s3 cp --only-show-errors - {dst}", check=check)
+    except:
+        command(f"aws s3 rm {dst}", check=False)
```

Comment:
If src doesn't exist, the above command would create an empty destination.   Work around that.   Perhaps we should also just check that src exists.